### DISCUSSION TITLE
markdown_it - texmath / dollarmath

### DISCUSSION THREAD

**Discussion by Uzay-G** [executablebooks/meta#202] [Q&A] [answered]
Hi,

I'm not sure this is the right place to ask, but I've been struggling to use the `texmath` and `dollarmath` extensions for markdown-it-py.

Example:

```python
>>> l = markdown_it.MarkdownIt().use(texmath.texmath_plugin)
>>> l.render("$x^1$")
'<p><eq>x^1</eq></p>\n'
```

Now this gives me some html with this `eq` tag, but the docs don't really explain how I am supposed to render that. Am I supposed to use `katex`?

Thanks a bunch, `markdown-it-py` looks really nice.

**Comment by chrisjsewell** [comment, marked as answer]
Heya, yes indeed you would need to use something like mathjax or katex to convert the "identified" math to rendered items.

I have only focussed on creating these plugins to generate the math syntax tokens, so that they can be passed to sphinx to do the actual rendering (via mathjax), and so haven't looked much into "direct" rendering I'm afraid.

As mentioned in the docs, `texmath` is directly derived from https://github.com/goessner/markdown-it-texmath, and you can see how that uses katex.
Ideally you would want a "pure-python" alternative to `texmath.katex.renderToString(tex, options)` (e.g. a complete version of https://github.com/tabatkins/katex),
or the math would need to be tagged in such a way that mathjax/katex can find it and convert it dynamically.

We welcome PRs, but likely won't be working on this ourselves any time soon.

**Comment by Uzay-G** [comment]
Ok thanks!
